snuba/clickhouse/query.py

```diff
@@ -102,3 +120,50 @@ def __init__(self,
     def format_sql(self) -> str:
         """Produces a SQL string from the parameters."""
         return self.__formatted_query
+
+
+class AstClickhouseQuery(ClickhouseQuery):
+    """
+    Clickhouse query that takes the content from the Snuba Query
+    AST and can be processed (through query processors) for Clickhouse
+    specific customizations.
+
+    Here the process of formatting the query, is independent from
+    the query body dictionary and it is performed starting from the
+    AST.
+    """
+
+    def __init__(self,
```

Comment:
Somewhere between the initialization of this class and the execution of format_sql, we need to run some Clickhouse query processing. This phase is relatively simple compared to the Snuba Query one and can either be triggered by the constructor itself or by the callsite that creates the object. I am not planning to solve this immediately because I want to get formatting working on this design first.